server/src/test/java/io/pravega/schemaregistry/storage/impl/group/GroupTest.java

```diff
@@ -0,0 +1,592 @@
+
+        List<TableRecords.TableValue> tableValueListVersionInfo = integerGroupTable.table.entrySet().stream().filter(
+                x -> x.getKey() instanceof TableRecords.VersionKey).map(
+                x -> x.getValue().getValue()).collect(
+                Collectors.toList());
+        assertFalse(tableValueListVersionInfo.isEmpty());
+        assertTrue(tableValueListVersionInfo.get(0) instanceof TableRecords.SchemaRecord);
+        TableRecords.SchemaRecord schemaRecord = (TableRecords.SchemaRecord) tableValueListVersionInfo.get(0);
+        assertEquals("mygroup", schemaRecord.getSchemaInfo().getType());
+        assertEquals(0, schemaRecord.getVersionInfo().getId());
+
+        List<TableRecords.TableValue> tableValueListSchemaInfo = integerGroupTable.table.entrySet().stream().filter(
+                x -> x.getKey() instanceof TableRecords.SchemaFingerprintKey).map(
+                x -> x.getValue().getValue()).collect(
+                Collectors.toList());
+        assertFalse(tableValueListSchemaInfo.isEmpty());
+        assertTrue(tableValueListSchemaInfo.get(0) instanceof TableRecords.SchemaVersionList);
+        TableRecords.SchemaVersionList schemaVersionValue =
+                (TableRecords.SchemaVersionList) tableValueListSchemaInfo.get(
+                        0);
+        assertEquals(0, schemaVersionValue.getVersions().get(0).getVersion());
+
+        List<TableRecords.TableValue> tableValueListLatestSchema = integerGroupTable.table.entrySet().stream().filter(
+                x -> x.getKey() instanceof TableRecords.LatestSchemaVersionForTypeKey).map(
+                x -> x.getValue().getValue()).collect(
+                Collectors.toList());
+        assertFalse(tableValueListLatestSchema.isEmpty());
+        assertTrue(tableValueListLatestSchema.get(0) instanceof TableRecords.LatestSchemaVersionValue);
+        TableRecords.LatestSchemaVersionValue latestSchemaVersionValue =
+                (TableRecords.LatestSchemaVersionValue) tableValueListLatestSchema.get(
+                        0);
+        assertEquals(0, latestSchemaVersionValue.getVersion().getVersion());
+
+        List<TableRecords.TableValue> tableValueListObjectTypesList =
+                integerGroupTable.table.entrySet().stream().filter(
+                        x -> x.getKey() instanceof TableRecords.SchemaTypesKey).map(
+                        x -> x.getValue().getValue()).collect(
+                        Collectors.toList());
+        assertFalse(tableValueListObjectTypesList.isEmpty());
+        assertTrue(tableValueListObjectTypesList.get(0) instanceof TableRecords.SchemaTypesListValue);
+        TableRecords.SchemaTypesListValue objectTypesListValue =
+                (TableRecords.SchemaTypesListValue) tableValueListObjectTypesList.get(
+                        0);
+        assertEquals(1, tableValueListObjectTypesList.size());
+        assertEquals("mygroup", objectTypesListValue.getTypes().get(0));
+    }
```

Comment:
There should be more cases here, esp failing cases -- 
1. multiple concurrent add schemas
2.1 schema type added for first time -- verification: version 0 assigned, object types entry created, latest version for type entry created.
2.2 schema type already present -- verification: higher version assigned, latest entry updated
2.3 add schema, add schema 2, delete schema, add schema 3 -- verify thata schema 3 gets correct version numbers.